snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedMemberOfFieldFixRequest.java

```diff
@@ -0,0 +1,252 @@
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.b2international.snowowl.core.domain.TransactionContext;
+import com.b2international.snowowl.core.events.Request;
+import com.b2international.snowowl.core.exceptions.ComponentNotFoundException;
+import com.b2international.snowowl.core.request.SearchResourceRequestIterator;
+import com.b2international.snowowl.snomed.common.SnomedTerminologyComponentConstants;
+import com.b2international.snowowl.snomed.core.domain.SnomedConcepts;
+import com.b2international.snowowl.snomed.core.domain.SnomedDescriptions;
+import com.b2international.snowowl.snomed.core.domain.refset.SnomedReferenceSet;
+import com.b2international.snowowl.snomed.core.domain.refset.SnomedReferenceSetMembers;
+import com.b2international.snowowl.snomed.datastore.index.entry.SnomedConceptDocument;
+import com.b2international.snowowl.snomed.datastore.index.entry.SnomedDescriptionIndexEntry;
+import com.b2international.snowowl.snomed.datastore.index.entry.SnomedRefSetMemberIndexEntry;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Sets;
+
+/**
+ * @since 7.21.1
+ */
+public class SnomedMemberOfFieldFixRequest implements Request<TransactionContext, Set<String>> {
+
+	private static final long serialVersionUID = 7930138235619925182L;
+	private static final int LIMIT = 10_000;
+	
+	Set<String> affectedComponentIds = new HashSet<>();
+	
+	@Override
+	public Set<String> execute(TransactionContext context) {
+		Logger log = LoggerFactory.getLogger("dataset-fix-SO-5690");
+		processConcepts(context, log);
+		processDescriptions(context, log);
+		return affectedComponentIds;
+	}
+	
+	private void processConcepts(TransactionContext context, Logger log) {		
+		Set<String> referenceSetIds = SnomedRequests.prepareSearchRefSet()
+				.filterByReferencedComponentType(SnomedTerminologyComponentConstants.CONCEPT)
+				.setLimit(Integer.MAX_VALUE)
+				.build()
+				.execute(context)
+				.stream()
+				.map(SnomedReferenceSet::getId)
+				.collect(Collectors.toSet());
+		
+		log.info("Found a total of " + referenceSetIds.size() + " concept type reference sets");
+
+		int counter = 0;
+		Multimap<String, String> missingMembersOfConcepts = HashMultimap.create();
+		Multimap<String, String> missingActiveMembersOfConcepts = HashMultimap.create();
+		
+		for (String referenceSetId : referenceSetIds) {
+			Multimap<String, String> missingMembersOf = HashMultimap.create();
+			Multimap<String, String> missingActiveMembersOf = HashMultimap.create();
+						
+			log.info("Processing batch " + counter++);
+			log.info("Processing refset " + referenceSetId);
+			
+			SnomedRefSetMemberSearchRequestBuilder memberRequest = SnomedRequests.prepareSearchMember()
+					.filterByRefSet(referenceSetId)
+					.setLimit(50_000)
+					.setFields("id", "active", SnomedRefSetMemberIndexEntry.Fields.REFERENCED_COMPONENT_ID);
```

Comment:
Use proper references to SnomedRf2Headers.

Reply:
or SnomedRefSetMemberIndexEntry.Fields.ID and SnomedRefSetMemberIndexEntry.Fields.ACTIVE